packages/core/src/local-workspace/remote_map/location_cache.ts

```diff
@@ -0,0 +1,88 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { logger } from '@salto-io/logging'
+import LRU from 'lru-cache'
+import { counters } from './counters'
+
+const log = logger(module)
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export class LocationCache extends LRU<string, unknown> {
+  readonly location: string
+
+  constructor(location: string, cacheSize: number) {
+    super({ max: cacheSize })
+    this.location = location
+  }
+}
+
+export type LocationCachePool = {
+  get: (location: string) => LocationCache
+
+  // The 'string' overload is temporary to allow the implementation of closeRemoteMapsOfLocation.
+  // Once we remove closeRemoteMapOfLocation, the 'string' overload should be removed.
+  release: (cache: LocationCache | string) => void
+}
+
+export type LocationCachePoolContents = Map<string, { cache: LocationCache; refcnt: number }>
+
+const DEFAULT_LOCATION_CACHE_SIZE = 5000
+
+export const createLocationCachePool = (
+  initialContents?: LocationCachePoolContents,
+  cacheSize: number = DEFAULT_LOCATION_CACHE_SIZE
+): LocationCachePool => {
+  // TODO: LRU if we determine too many locationCaches are created.
+  const pool: LocationCachePoolContents = initialContents ?? new Map<string, { cache: LocationCache; refcnt: number }>()
```

Comment:
I think typescript will allow this so you don't have to repeat the type
```suggestion
  const pool: LocationCachePoolContents = initialContents ?? new Map()
```